be/src/exec/vectorized/hash_join_node.cpp

```diff
@@ -179,21 +179,25 @@ Status HashJoinNode::open(RuntimeState* state) {
         }
 
         {
-            RETURN_IF_ERROR(state->check_mem_limit("HashJoinNode"));
             // copy chunk of right table
             SCOPED_TIMER(_copy_right_table_chunk_timer);
             try {
                 RETURN_IF_ERROR(_ht.append_chunk(state, chunk));
+                RETURN_IF_ERROR(state->check_mem_limit("HashJoinNode"));
             } catch (std::bad_alloc const&) {
                 return Status::MemoryLimitExceeded("Mem usage has exceed the limit of BE");
             }
         }
     }
 
     {
-        // build hash table: compute key columns, and then build the hash table.
-        RETURN_IF_ERROR(_build(state));
-        RETURN_IF_ERROR(state->check_mem_limit("HashJoinNode"));
+        try {
+            // build hash table: compute key columns, and then build the hash table.
+            RETURN_IF_ERROR(_build(state));
+            RETURN_IF_ERROR(state->check_mem_limit("HashJoinNode"));
```

Comment:
This pr is check memory limit after real memory allocate?

Reply:
Currently, most  is allocated first and then checked but large memory are checked first and then allocated in mem_hook

https://github.com/StarRocks/starrocks/pull/1922/files

Reply:
check after one chunk append or build ht finished